subsys/logging/log_msg.c

```diff
@@ -358,6 +358,10 @@ void z_log_msg_runtime_vcreate(uint8_t domain_id, const void *source,
 		plen = 0;
 	}
 
+	if (plen >= (1 << Z_LOG_MSG_PACKAGE_BITS)) {
```

Comment:
I added it for static creation (https://github.com/zephyrproject-rtos/zephyr/blob/83fe779a4c36d910e853403af79e4b091b46ba1e/subsys/logging/log_msg.c#L307) but forgot to handle runtime. Can you apply same approach as it is there. It's better to drop that message. Decoding of incomplete message will fail anyway.